# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/sessionstore/SessionStoreUtils.cpp
# Commit: a30f158eba45
# Full Hash: a30f158eba4511313be684c4b5f3d5f278f34103
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-04-22 21:48:48
# Regressor Bug: 1580565
# File Overlap Count: 3
# Description:
#   Bug 1580565 - Part 4: Use WindowContext to manage BrowsingContext cached status, r=farre
#   
#   The existing infrastructure which stored cached BrowsingContexts on the
#   BrowsingContextGroup was added before WindowContexts were added, and can cause
#   racing issues with partially discarded trees during process switches.
# ==============================================================================

diff -r 276b131190a8 -r a30f158eba45 toolkit/components/sessionstore/SessionStoreUtils.cpp
--- a/toolkit/components/sessionstore/SessionStoreUtils.cpp	Wed Apr 22 03:21:55 2020 +0000
+++ b/toolkit/components/sessionstore/SessionStoreUtils.cpp	Wed Apr 22 06:05:41 2020 +0000
@@ -1169,7 +1169,7 @@
   }
 
   // This is not going to work for fission. Bug 1572084 for tracking it.
-  for (BrowsingContext* child : aBrowsingContext->GetChildren()) {
+  for (BrowsingContext* child : aBrowsingContext->Children()) {
     window = child->GetDOMWindow();
     if (!window) {
       return;
@@ -1279,7 +1279,7 @@
   uint32_t trailingNullCounter = 0;
 
   // This is not going to work for fission. Bug 1572084 for tracking it.
-  for (auto& child : aBrowsingContext->GetChildren()) {
+  for (auto& child : aBrowsingContext->Children()) {
     NullableRootedDictionary<CollectedData> data(aCx);
     CollectFrameTreeData(aCx, child, data, aFunc);
     if (data.IsNull()) {
